### PR TITLE
remove locale "sd" (not supported by ICU/CLDR); unify localizations script options

### PR DESCRIPTION
**Pull request by yjbanov**
- Remove locale "sd", which is not supported by ICU/CLDR (see the banner at the top of this page: http://demo.icu-project.org/icu-bin/locexp?d_=en&_=sd)
- Unify the CLI options for `gen_localizations.dart` and `gen_date_localizations.dart`.

### CODE REVIEW

**Comment by HansMuller** [comment]
LGTM modulo a few nits and --overwrite :-)

**Comment by yjbanov** [comment]
@HansMuller did you forget to send the nits?

File: dev/tools/gen_localizations.dart [hunk 1]
```diff
@@ -135,7 +151,15 @@ void main(List<String> args) {
...
-  print(outputHeader.replaceFirst('@(regenerate)', regenerate));
-  print(generateLocalizationsMap());
+  final String regenerate = 'dart dev/tools/gen_localizations.dart -w';
+  final StringBuffer buffer = new StringBuffer();
+  buffer.writeln(outputHeader.replaceFirst('@(regenerate)', regenerate));
+  buffer.writeln(generateLocalizationsMap());
+
+  if (options.writeToFile) {
+    final File localizationsFile = new File(pathlib.join(directory.path, 'localizations.dart'));
+    localizationsFile.writeAsStringSync('$buffer');
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply dev/tools/gen_localizations.dart]: That's actually idiomatic in Dart. Search for `\'\$\w+\'` in our codebase, you'll find lots.

File: dev/tools/gen_localizations.dart [hunk 2]
```diff
@@ -2,30 +2,47 @@
...
 
-// Given a directory that contains localized ".arb" (application resource
-// bundle) files, generates a Dart "localizations" Map definition that combines
+// This program generates a Dart "localizations" Map definition that combines
 // the contents of the arb files. The map can be used to lookup a localized
-// string: localizations[localeString][resourceId].
+// string: `localizations[localeString][resourceId]`.
 //
-// See *.arb and localizations.dart in packages/flutter/lib/src/material/i18n/.
+// The *.arb files are in packages/flutter/lib/src/material/i18n/.
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply dev/tools/gen_localizations.dart]: Good catch! Done.

File: dev/tools/gen_localizations.dart [hunk 3]
```diff
@@ -2,30 +2,47 @@
...
+// Run this program from the root of the git repository.
+//
+// The following outputs the generated Dart code to the console as a dry run:
+//
+// ```
+// dart dev/tools/gen_localizations.dart
+// ```
+//
+// If the data looks good, use the `-w` option to overwrite the
+// lib/src/l10n/localizations.dart file:
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply dev/tools/gen_localizations.dart]: Done.

File: dev/tools/gen_localizations.dart [hunk 4]
```diff
@@ -2,30 +2,47 @@
...
+//
+// ```
+// dart dev/tools/gen_localizations.dart -w
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply dev/tools/gen_localizations.dart]: Done.

File: dev/tools/gen_localizations.dart [hunk 5]
```diff
@@ -135,7 +151,15 @@ void main(List<String> args) {
   }
   validateLocalizations(localeToResources, localeToResourceAttributes);
 
-  final String regenerate = 'dart dev/tools/gen_localizations.dart ${directory.path} ${args[1]}';
-  print(outputHeader.replaceFirst('@(regenerate)', regenerate));
-  print(generateLocalizationsMap());
+  final String regenerate = 'dart dev/tools/gen_localizations.dart -w';
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply dev/tools/gen_localizations.dart]: Done.

**Pull request merged**